### ISSUE TITLE
[BUG] Guard's default clothing and given armour clip together

### ISSUE THREAD

**Issue by slubbery** [ldtteam/minecolonies#9740]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Are you using the latest MineColonies Version?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [X] Single Player
- [ ] Multi Player

### Minecraft Version

1.20.1

### MineColonies Version

1.20.1 - 1.1.451-BETA

### Structurize Version

1.20.1 - 1.0.686-BETA

### Related Mods and their Versions

1. forge: 1.20.1-forge-47.2.0
2. blockui: 1.20.1 - 1.0.134 - BETA
3. domum ornamentum: 1.20 - 1.0.153 - BETA

### Current Behavior

![image](https://github.com/ldtteam/minecolonies/assets/157757635/835cb1df-2267-43be-9d46-313a1b60a701)
basically when i give a guard armour, the guard's default clothing does not disappear. instead it clips with the armour i gave them which doesn't look very aesthetically pleasing! note that this bug is present for all armour pieces, not just the helmet like in the image

### Expected Behavior

the default outfit of the guard should disappear, and they should wear the given armour

### Reproduction Steps

1. give armour to guard
2. Enjoy!

### Logs

https://pastebin.com/q4SWVhme

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by hjhhhc** [comment]
I also have this situation
